src/storage/src/table/state_table.rs

```diff
@@ -25,43 +25,70 @@ use risingwave_common::util::ordered::{serialize_pk, OrderedRowSerializer};
 use risingwave_common::util::sort_util::OrderType;
 use risingwave_hummock_sdk::key::range_of_prefix;
 
-use super::cell_based_table::CellBasedTable;
+use super::cell_based_table::CellBasedTableExtended;
 use super::mem_table::{MemTable, RowOp};
+use crate::cell_based_row_serializer::CellBasedRowSerializer;
+use crate::cell_serializer::CellSerializer;
 use crate::error::{StorageError, StorageResult};
 use crate::{Keyspace, StateStore};
 
 /// `StateTable` is the interface accessing relational data in KV(`StateStore`) with encoding.
+pub type StateTable<S> = StateTableExtended<S, CellBasedRowSerializer>;
+
+impl<S: StateStore> StateTable<S> {
+    pub fn new(
+        keyspace: Keyspace<S>,
+        column_descs: Vec<ColumnDesc>,
+        order_types: Vec<OrderType>,
+        dist_key_indices: Option<Vec<usize>>,
+        pk_indices: Vec<usize>,
+    ) -> Self {
+        StateTableExtended::new_extended(
+            keyspace,
+            column_descs,
+            order_types,
+            dist_key_indices,
+            pk_indices,
+            CellBasedRowSerializer::new(),
+        )
+    }
+}
+
+/// `StateTableExtended` is the interface accessing relational data in KV(`StateStore`) with
+/// encoding, using `CellSerializer` for row to cell serializing.
 #[derive(Clone)]
-pub struct StateTable<S: StateStore> {
+pub struct StateTableExtended<S: StateStore, SER: CellSerializer> {
```

Comment:
I‘m not sure is this by design? 
Maybe I missed some of the discussion, is there any RFC about this?